spec/features/idv/in_person_spec.rb

```diff
@@ -338,18 +332,27 @@
       expect(@sms_link).to be_present
 
       perform_in_browser(:mobile) do
+        # doc auth page
         visit @sms_link
         mock_doc_auth_attention_with_barcode
         attach_and_submit_images
 
+        # error page
         click_button t('in_person_proofing.body.cta.button')
-        search_for_post_office
-        within page.first('.location-collection-item') do
-          click_spinner_button_and_wait t('in_person_proofing.body.location.location_button')
-        end
 
+        # location page
+        expect(page).to have_content(t('in_person_proofing.headings.po_search.location'))
+        complete_location_step
+
+        # prepare page
+        # the 10s wait is added because the prepare page isn't
+        # consistently loading within the default wait time
```

Comment:
This seems concerning / worth understanding. Is there some delay that happens in the browser outside of the spinner button completing, since it does look like `complete_location_step` helper is waiting for that spinner to go away.

I think the changes in `search_for_post_office` make a lot of sense to wait for the search to finish. Was that change alone not enough to fix all of the flakiness?

Reply:
I think it's [the call to the backend to save the selected location](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/document-capture/components/in-person-location-post-office-search-step.tsx#L68-L71) that's pushing the delay past the default wait threshold. In a82513e9a3a823be0178732112968900cdaac262 I've tried moving the wait into `complete_location_step`, which is slightly better, but maybe we should look at making that location save asynchronous.

Reply:
But the spinner would be showing for a user at that point, right? `click_spinner_button_and_wait` should be waiting for the spinner to go away, which in this case I think would only happen once that back-end save has finished. Or at least that's how I'd imagine it _should_ be working. If there's a delay happening beyond that button click, it'd be nice if we could either make sure that happens while the spinner is spinning, or provide some other user-facing feedback that we can use as an indicator in the tests.

Reply:
okay, part of the issue is that the location select buttons aren't spinning when clicked. I've created [LG-9518](https://cm-jira.usa.gov/browse/LG-9518) for that bug. In the meantime I'd like to merge this fix so I can unblock some other work.